Search/ReIndex/ResumeManager.php

```diff
@@ -0,0 +1,125 @@
+<?php
+/*
+ * This file is part of the MassiveSearchBundle
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Massive\Bundle\SearchBundle\Search\ReIndex;
+
+/**
+ * Allow the storage of "checkpoints" which indicate how far
+ * a reindexing process has progressed in the case that it is
+ * interupted (f.e. by an error) with the aim of allowing the
+ * reindexing process to resume from where it left off.
+ *
+ * This is basically a key/value store.
+ */
+class ResumeManager
+{
+    /**
+     * Store a checkpoint. This would typically be an offset
+     * from where the reindexer can subsequently resume its
+     * task.
+     *
+     * @param mixed $name
+     * @param mixed $value
+     */
+    public function setCheckpoint($name, $value)
```

Comment:
The Sulu Structure Reindexer has this class injected into it.

On Fri, Feb 12, 2016 at 04:12:55AM -0800, Daniel Rotter wrote:
>    In [1]Search/ReIndex/ResumeManager.php:
> 
>  > + * interupted (f.e. by an error) with the aim of allowing the
>  > + * reindexing process to resume from where it left off.
>  > + *
>  > + * This is basically a key/value store.
>  > + */
>  > +class ResumeManager
>  > +{
>  > +    /**
>  > +     * Store a checkpoint. This would typically be an offset
>  > +     * from where the reindexer can subsequently resume its
>  > +     * task.
>  > +     *
>  > +     * @param mixed $name
>  > +     * @param mixed $value
>  > +     */
>  > +    public function setCheckpoint($name, $value)
> 
>    Why does Sulu have to know about this?
> 
>    —
>    Reply to this email directly or [2]view it on GitHub.
> 
>    Reverse link: [3]unknown
> 
> References
> 
>    Visible links
>    1. https://github.com/massiveart/MassiveSearchBundle/pull/74#discussion_r52734341
>    2. https://github.com/massiveart/MassiveSearchBundle/pull/74/files#r52734341
>    3. https://github.com/massiveart/MassiveSearchBundle/pull/74/files#r52734341